scanners/seo.py

```diff
@@ -56,46 +81,68 @@ def init(environment: dict, options: dict) -> dict:
+        'Total URLs': sitemap_results['url_tag_count'] if sitemap_results['url_tag_count'] else 0,
+        'Est time to index': 'Unknown',
+        'Main tags found': False,
+        'Search found': False,
+        'Warnings': {},
     }
-    # Now let's take a look at sitemap
-    sitemap = requests.get("https://" + domain + '/sitemap.xml')
-    sitemap_status = sitemap.status_code
-    results['sitemap'] = {
-        'status': sitemap_status
-    }
-    # If we have found a sitemap, see how many URLs are in there?
-    # TO DO: If sitemap.xml redirects to a sitemap index,
-    # we're not handling that correctly yet. We need to detect that
-    # redirect, then ... what? Follow from there?
-    if sitemap_status == HTTPStatus.OK:
-        soup = BeautifulSoup(sitemap.text, 'xml')
-        urls = soup.find_all('url')
-        results['sitemap']['urls found'] = len(urls)
-        # and how many of those URLs appear to be PDFs
-        if urls:
-            pdfcount = len([u for u in urls if '.pdf' in u.get_text()])
-            results['sitemap']['PDFs found in sitemap'] = pdfcount
+
+    # See if we can determine platforms used for the site
+    build_info = builtwith(fqd)
+    if 'web-frameworks' in build_info:
+        results['Platforms'] = build_info['web-frameworks']
+
+    # If we found additional sitemaps in a sitemap index or in robots.txt, we
+    # need to go look at them and update our url total.
+    additional_urls = 0
+    for loc in sitemap_results['sitemap_locations_from_index']:
+        if loc != sitemap_results['final_url']:
+            sitemap = requests.get(loc)
+            if sitemap.status_code == HTTPStatus.OK:
+                soup = BeautifulSoup(sitemap.text, 'xml')
+                additional_urls += len(soup.find_all('url'))
+
+    for loc in sitemap_results['sitemap_locations_from_robotstxt']:
+        if loc != sitemap_results['final_url']:
+            sitemap = requests.get(loc)
+            if sitemap.status_code == HTTPStatus.OK:
+                soup = BeautifulSoup(sitemap.text, 'xml')
+                additional_urls += len(soup.find_all('url'))
+    results['Total URLs'] = results['Total URLs'] + additional_urls
+
+    # Can we compute how long it will take to index all URLs (in hours)?
+    if results['Crawl delay']:
+        results['Est time to index'] = (int(results['Total URLs']) * int(results['Crawl delay'])) / 3600
 
     # We'll write to these empty lists for simple dupe checking later
     titles = []
     descriptions = []
-    # Perform the "task".
     for page in environment['pages']:
         try:
             r = requests.get("https://" + domain + page, timeout=4)
             # if we didn't find the page, write minimal info and skip to next page
             if r.status_code != HTTPStatus.OK:
-                results[page] = {
-                    'page': page,
-                    'status': str(r.status_code)
-                }
+                results[page] = '404'
```

Comment:
Basically, because it's easier to sort on the output side and anything other than a 200 is, at least as far as the business purpose goes, the same thing.

Reply:
Would there be value in knowing if it's an intermittent error vs. a permanent change?

Reply:
Possibly, in the future if/when we start pulling pages from nav. for now, with the two hard-coded paths, I think it's fine to just fail fast and move on.